server/app/middlewares/websocket_backend.rb

```diff
@@ -246,6 +246,7 @@ def our_version
   # @param [HostNode] node
   def handle_invalid_agent_version(ws, node)
     node.set(connected: false, last_seen_at: Time.now.utc)
+    # XXX: delay to give the Agent::NodePlugger.send_master_info -> MongoPubSub time to call send_message before the websocket gets closed... hopefully?
     EventMachine::Timer.new(1) do
       ws.close(4010) if ws
```

Comment:
If I'm reading this right, this feels pretty evil... if the `Agent::NodePlugger.send_master_info  -> `MongoPubSub` is slow, the websocket might get closed before the `/agent/master_info` RPC gets sent.